### ISSUE TITLE
Poetry publishes package under different name

### ISSUE THREAD

**Issue by eblis** [python-poetry/poetry#4631]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x ] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.

- **Windows 10 x64 **: 
- **Poetry version 1.1.11**: 
- Pyproject.toml
```
[tool.poetry]
name = "peds_common"
version = "0.0.6"
description = "PEDS common libraries"
packages = [
	{ include = "peds" }
]
include = [ "README.md" ]

[[tool.poetry.source]]
name = 'peds'
url = '<artifactory url>/PEDSSW_pypi'

[tool.poetry.dependencies]
python = "^3.9"
pyaml = "^21.10.1"
cachetools = "^4.2.4"
colorlog = "^6.5.0"
colorama = "^0.4.4"
cryptography = "^35.0.0"

[tool.poetry.dev-dependencies]
pytest = "^6.2.5"

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

## Issue
Poetry says that it will deploy the project named `peds_common`, but it uploads to `peds-common` in Artifactory.
`poetry publish` command
![image](https://user-images.githubusercontent.com/540188/137088740-f37a7dc9-b45e-4e1b-ba29-b8244afb886c.png)

package name in artifactory:
![image](https://user-images.githubusercontent.com/540188/137089086-af5b46bc-237c-478e-9648-049c94f4dddf.png)

The version uploaded by `poetry` is `0.0.6` under `peds-common`, the other ones uploaded under `peds_common` are uploaded by setup.py before moving to `poetry`

**Comment by Mariatta** [comment]
PEP 503 states that project names should be [normalized](https://www.python.org/dev/peps/pep-0503/#normalized-names) as follows:
> lowercased with all runs of the characters ., -, or _ replaced with a single - character. 

If I understand the code correctly, `poetry-core` normalized your package name to `peds-common`.

https://github.com/python-poetry/poetry-core/blob/1ad005fcb371710c36bbead6cbdd14e6d7f8a948/poetry/core/utils/helpers.py#L26-L27, therefore when you use poetry to publish, it normalized the package name to `peds-common` 

Perhaps setuptools did not normalize the package name per PEP 503, and therefore it uploaded to the `peds_common` instead of `peds-common`. 

So I'm not sure if this is a bug. Perhaps needing some documentation/faq regarding this behavior.

**Comment by shanelanan** [comment]
Is there anyway to override this behavior?  I am trying to publish to an existing repository that I do not have control over.  It was published with underscores from an old `setup.py` script.

**Comment by jack-mcivor** [comment]
Is this still valid? It seems to me like:

- The name field of the [tool.poetry] section is respected for the upload path (hyphens stay hyphens, underscores stay underscores)
- This is the version referenced in the output of `poetry build`
- The built wheel and sdist replace hyphens with underscores in the package name
- `poetry version` always replaces underscores with hyphens

**Comment by radoering** [comment]
Should be resolved by python-poetry/poetry-core#620
